### PR TITLE
KS Emoticon Glyphs 5.4

### PR DESCRIPTION
**Pull request by alex40724**
Feature Request:
https://www.ilias.de/docu/goto_docu_wiki_wpage_4698_1357.html

Rules:
https://github.com/ILIAS-eLearning/ILIAS/pull/908/files

### CODE REVIEW

**Comment by alex40724** [comment]
Two questions:
- Any idea why this is not appearing in the KS docs with these code changes? What is missing?
- Is this the way to proceed with the other emoticon glyphs?

File: src/UI/Component/Glyph/Factory.php [hunk 1]
```diff
@@ -499,4 +499,51 @@ public function note($action = null);
...
 	 */
 	public function comment($action = null);
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       Clicking the Like Glyph indicates a user affirms an item, e.g. a posting.
+	 *   composition: >
+	 *       The Like Glyph uses the "thumbs up" unicode emoji U+1F44D
```
Comment (Amstutz) [inline src/UI/Component/Glyph/Factory.php]: Add Period. Any hints about this "special unicode", any reference. Ist this some kind of standard. Maybe note, that this is not from the default bootstrap set, which is fine as long as we have a JF dec. for that.

File: src/UI/Component/Glyph/Factory.php [hunk 2]
```diff
@@ -499,4 +499,51 @@ public function note($action = null);
...
+	 *   effect: >
+	 *       Clicking toggles the state of the user expression. Clicking also updates the respective counter.
```
Comment (Amstutz) [inline src/UI/Component/Glyph/Factory.php]: Is it possible to somehow add a hover effect to this? Currently there is none.

File: src/UI/Component/Glyph/Factory.php [hunk 3]
```diff
@@ -499,4 +499,51 @@ public function note($action = null);
...
+	 */
+	public function like($action = null);
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       Clicking the Love Glyph indicates a user adores an item, e.g. a posting.
+	 *   composition: >
+	 *       The Love Glyph uses the "red heart" unicode emoji U+2764
```
Comment (Amstutz) [inline src/UI/Component/Glyph/Factory.php]: Same as above, period, reference etc.

**Review by Amstutz — CHANGES_REQUESTED** [review]
Thanks for this commit. I commented on some rather minor details. Note that I would also love to have a "dislike" emo. However, I do not have a strong opinion on this.

Concerning your questions:
- Worked like a charm on my local installation. Did you click on the "Refresh Entries"-Button? Note that this button only appears in Dev-Mode and if you activated the manage system style setting in the setup.
- Yes, this seems correct to me. After all, the def. by using unicodes is what makes them glyphs and not icons or graphical buttons to me.

**Comment by klees** [comment]
https://github.com/MorbZ/OpenSansEmoji is an Emoji-Font with Apache 2.0 License. Don't know how good it is, though.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 12 MAR 2018: We accept this PR for trunk. Please introduce monochrome glyphs for all needed emoticons.

**Comment by Amstutz** [comment]
We would like to help working out a "ILIAS" default Glyph-Font set. Contact us. Note that: "All Glyphs MUST be taken from the Bootstrap Glyphicon Halflings set. Exceptions MUST be approved by the JF." So it is oke, to propose new Glyphs.

**Pull request merged**